### PR TITLE
Add grid system and grid story to Prisma

### PR DESCRIPTION
**Pull request by lcsrinaldi**

### CODE REVIEW

File: stories/Grid/index.story.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+//   'grid with two span 1 columns',
+//   withInfo()(() => (
+//   )),
+// );
+
+// stories.add(
+//   'grid with inner grid',
+//   withInfo()(() => (
+//   )),
+// );
```
Comment (noxan) [inline stories/Grid/index.story.jsx]: What about these stories?

Comment (lcsrinaldi) [reply stories/Grid/index.story.jsx]: on 🔥  again? was not finished :) now I'm

**Comment by lcsrinaldi** [comment]
@noxan, I see, for the sake of organizing myself I'll address this in points.

1. I agree with you about the property names, they are somewhat confusing, I guess it was a design decision to make everything compact (hence the components `Box`, `Grid`, `Flex`). Although I must add that with a couple of days messing around with this library I just don't have anymore problems in using them. `m`/`p` for padding and margin, `t`/`r`/`b`/`l` for directions and `x`/`y` for x (left/right) and y (top/bottom) axis. To me it's now really straightforward.

2. About the library's flexibility, before I chose this one (by Gabriel's indication and comparing with [this list](https://github.com/styled-components/awesome-styled-components#grid-systems)) I had a look into other libraries and I tell you it's hard to find one that do a good job in constraining you and that fits into our design system, one of the points of choosing this one was about flexibility, so we can fit any grid that the design team decides in it.
And although this library is flexible, it can enforces good constraints regarding components spacing and breakpoints, it does a really good job with it (we can explore this further)

3. IMO it's not the library's responsibility to constrain the users of Prisma, but ours. For one, we can decide to not export this grid components to our applications and only use them inside `Prisma`, which will have a more thorough filter and review of code that we insert in. Later on if we decide that we need some layout components inside our applications we can build them on top of the library and export to our applications.

4. Yes this `width` field takes percentages, css values in strings and pixel values, it's really flexible, though I don't see that as a bad thing, since we can interface the use of this component with something like:

```javascript
const Column = ({ span, theme, children }) => (
  <Box width={span / theme.gridSize}>{children}</Box>
);
```

Using `Column` and `span` just to do a parallel with our old grid, although I like the semantics of taking one half of the page (`1 / 2`) or one third (`1 / 3`), or one twelfth (`1 / 12`), which is used by grid-styled.

 ####  Conclusion

So, to sum it up, those a very valid points, but I think most of them are not going to be addressable just by changing libraries, it's going to be our library responsibility. 
If you want we can do a more thorough benchmark to compare other grid libraries. Up to you and the other guys. Thanks for your input 👍

**Comment by lcsrinaldi** [comment]
@noxan, you are totally right, they sure seem like a documentation for Prisma users. I should've said something that my intent with this was only to replicate our old grid stories with the new grid system  and later we improve starting from this point. And even now that you touched this subject I realize that the section saying we use grid-styled (Grid > Documentation) should only exist on Prisma's Github documentation and not on our Storybook since we are not going to export it.
